### ISSUE TITLE
Fatal error sending order complete email for free local pickup orders

### ISSUE THREAD

**Issue by aqdennis** [myparcelnl/woocommerce#829]
### Plugin version

4.10.2

### WooCommerce version

6.3.1

### WordPress version

5.9.3

### PHP version

7.3

### What went wrong?

- For orders with local postal codes we offer free pickup, in which case we do not have to export to MyParcel and thus the order does not have a track-and-trace code.
- In our "order complete" emails we include the MyParcel track and trace  codes
- If from the backend order list I click the  checkmark button behind the order with local pickup to set it to "complete" we receive a fatal error. Apparently it tries to add the track-and-trace number to the email, but this order doesn't have a track-and-trace number because the customer chose a local pickup.
- If from the backend order list I click on the order to see the order overview and from there I send the "order complete" email to the customer it works fine. Hence, it's just from the order list that the fatal error occurs.

Somehow it seems to think this is a MyParcel order. But apart from that....it should check whether there is a track and trace number before adding it to the email.

### Reproduction steps

1. Create an order with local pickup - probably any shipping method that is not MyParcel would work.
2. Go to the back end order list (WooCommerce -> orders)
3. Click on the checkmark button behind the order to set the order as complete

![image](https://user-images.githubusercontent.com/48013601/165970199-576689c6-49bf-415a-84e8-c017516a3b46.png)

### Relevant log output

```shell
2022-04-29T14:30:32+00:00 CRITICAL Uncaught TypeError: Argument 2 passed to WCMP_Frontend_Track_Trace::addTrackTraceToEmail() must be of the type bool, null given, called in public_html/shop/wp-includes/class-wp-hook.php on line 309 and defined in public_html/shop/wp-content/plugins/woocommerce-myparcel/includes/frontend/class-wcmp-frontend-track-trace.php:38
Stack trace:
#0 public_html/shop/wp-includes/class-wp-hook.php(309): WCMP_Frontend_Track_Trace->addTrackTraceToEmail(Object(WC_Order), NULL)
#1 public_html/shop/wp-includes/class-wp-hook.php(331): WP_Hook->apply_filters('', Array)
#2 public_html/shop/wp-includes/plugin.php(474): WP_Hook->do_action(Array)
#3 public_html/shop/wp-content/themes/deli/woocommerce/emails/email-order-details.php(22): do_action('woocommerce_ema...', Object(WC_Order), NULL, NULL, NULL)
#4 publi in public_html/shop/wp-content/plugins/woocommerce-myparcel/includes/frontend/class-wcmp-frontend-track-trace.php op de lijn 38
```

**Comment by aqdennis** [comment]
Apparently the problem is that when the action is fired to create the email the "send_to_admin" variable is NULL while the hook function "addTrackTraceToEmail" explicitly defined the second argument as Bool.
Question is what is wrong....the fact that argument is NULL or the fact that the hook function explicitly wants a Bool. Probably typecasting the arguments when do_action is called should do the trick, but that would be on the WooCommerce side and not on the MyParcel side of the problem.

**Comment by wthijmen** [comment]
Hi @aqdennis,
This issue is fixed since release [v4.13.0](https://github.com/myparcelnl/woocommerce/releases/tag/v4.13.0)
